### PR TITLE
chore(cli): teamsfx upgrade

### PR DESCRIPTION
**Pull request by Alive-Fish**
https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/16625425
https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/16625428

When `TEAMSFX_V3="true"`, `teamsfx -h` will show (`teamsfx upgrade` is added):
```bash
C:\Users\zhiyou\source\zhiyouTestTab>teamsfx -h
teamsfx <command>

Commands:
  teamsfx account <action>               Manage cloud service accounts. The supported cloud services
                                         are 'Azure' and 'Microsoft 365'.
  teamsfx new                            Create a new Teams application.
  teamsfx provision                      Provision the cloud resources in the current application.
  teamsfx deploy                         Deploy the current application.
  teamsfx package                        Build your Teams app into a package for publishing.
  teamsfx validate                       Validate the Teams app manifest.
  teamsfx publish                        Publish the app to Teams.
  teamsfx config <action>                Configure user settings.
  teamsfx preview                        Preview the current application.
  teamsfx env <action>                   Manage environments.
  teamsfx permission <action>            Check, grant and list user permission.
  teamsfx init <part>                    Initialize the project for using Teams Toolkit.
  teamsfx update <application-manifest>  Update the specific application manifest file.
  teamsfx upgrade                        Upgrade the project to the latest version of Teams Toolkit.

Options:
      --verbose      Print additional information.                        [boolean] [default: false]
      --debug        Print diagnostic information.                        [boolean] [default: false]
      --interactive  Run the command interactively.                                        [boolean]
  -h, --help         Show help                                                             [boolean]
  -v, --version      Show version number                                                   [boolean]

For more information about the Teams Toolkit - https://aka.ms/teamsfx-cli.
```

`teamsfx upgrade -h` will show: (adding the `--force` option to decide whether ask user to confirm again) 
```bash
C:\Users\zhiyou\source\zhiyouTestTab>teamsfx upgrade -h
teamsfx upgrade

Upgrade the project to the latest version of Teams Toolkit.

Options:
      --verbose  Print additional information.                            [boolean] [default: false]
      --debug    Print diagnostic information.                            [boolean] [default: false]
  -h, --help     Show help                                                                 [boolean]
      --force    Force to upgrade the project to the latest version of Teams Toolkit.      [boolean]
```

We will show an error if users run commands with `--interactive false` in a V2 project (this will let users not upgrade automatically in the CICD scenario):
```bash
C:\Users\zhiyou\source\zhiyouTestTab>teamsfx provision --interactive false
(×) [TeamsfxCLI.NotAllowedMigrationError]: Teams toolkit's pre-released version supports new project configuration and is incompatible with previous versions. Please try it by creating a new project or run "teamsfx upgrade" to upgrade your project first.
```

Users can run `teamsfx upgrade` to manually upgrade the project: (using `--force` will not ask users for confirmation)
```bash
C:\Users\zhiyou\source\zhiyouTestTab>teamsfx upgrade
? Please upgrade your project to stay compatible with the latest version, your current project contains configurations from an older Teams Toolkit.
The auto-upgrade process will generate backups in case an error occurs.: (Use arrow keys)
(*) Learn more
( ) Upgrade
( ) Cancel

C:\Users\zhiyou\source\zhiyouTestTab>teamsfx upgrade --force
Upgrade project successfully.

```

### CODE REVIEW

File: packages/cli/src/resource/strings.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "error": {
+    "NotAllowedMigrationErrorMessage": "Teams toolkit's pre-released version supports new project configuration and is incompatible with previous versions. Please try it by creating a new project or run \"teamsfx upgrade\" to upgrade your project first."
```
Comment (sffamily) [inline packages/cli/src/resource/strings.json:3]: Teams toolkit -> Teams Toolkit
pre-released -> Pre-Release
remove "Please" and try -> Try

File: packages/cli/src/resource/strings.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  },
+  "command": {
+    "upgrade": {
+      "description": "Upgrade the project to the latest version of Teams Toolkit.",
+      "options": {
+        "force": "Force to upgrade the project to the latest version of Teams Toolkit."
```
Comment (sffamily) [inline packages/cli/src/resource/strings.json:9]: Force to upgrade -> Force upgrade

optional:   to the latest -> to work with the latest

File: packages/cli/src/resource/strings.json (hunk 1 above)
Comment (Alive-Fish) [reply packages/cli/src/resource/strings.json]: Thank you, and resolved.

File: packages/cli/src/resource/strings.json (hunk 2 above)
Comment (Alive-Fish) [reply packages/cli/src/resource/strings.json]: Thank you and resolved.

**Review by sffamily — APPROVED** [review]

**Review by chagong — APPROVED** [review]

**Pull request merged**